### DISCUSSION TITLE
Linux: External tablet input is ignored on Wayland unless raw input is re-set to off

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#30980] [Q&A]
The in-game tablet implementation doesn't allow the usage of extra filter plugins in OpenTabletDriver. So in order to have interpolation/anti-chatter, you have to disable the in-game tablet support. Now tablet input doesn't work at all! 

I have a very annoying workaround to get tablet working:

1. Launch the game with `SDL_VIDEODRIVER=wayland`
2. Now the game will start in the correct resolution, but not actually fullscreen despite the game being set to fullscreen already. At this point I press ALT-ENTER three times so that it goes to windowed mode -> borderless -> fullscreen. Now it's properly fullscreen.
3. Open the settings, and toggle "High Precision Mouse" from off to on, then to off again. Now the tablet works as expected..

So basically you need to use the Wayland SDL video driver, then double-toggle "High Precision Mouse" in order for tablet to work. It works, but it's annoying and you have to do it every time you start up the game. I almost wanted to fork this repository and add my own code that re-sets those two settings after launching the game but then I can't play online.

Is there a nicer way around this?

I've tested this natively (built from repository), with Flatpak, AppImage as well as the AUR package for osu-lazer-bin. Using Arch Linux and OpenTabletDriver (Wacom CTL-472). KDE Plasma (Wayland session).

**Comment by mrdandelion6** [comment]
i tried this but it didn't work for me. i have same set up as you , using osu-lazer-bin from aur:
```sh
osu-lazer SDL_VIDEODRIVER=wayland
```

then i toggle into full screen back by 3 alt enters , and also toggle high precision.. but nothing. doesn't work. i'm using wacom CTL-4100WL with no filters.

**Reply by mrdandelion6** [reply]
oh shit i fixed , had to do 
```sh
env SDL_VIDEODRIVER=wayland osu-lazer
```

lol
